File: TDDMicroExercises/PHP/TirePressureMonitoringSystem/Sensor.php
<?php

namespace TDDMicroExercises\PHP\TirePressureMonitoringSystem;

class Sensor 
{
	const OFFSET = 16;

	public static function SamplePressure()
	{
		// placeholder implementation that simulate a real sensor in a real tire
		$pressureTelemetryValue = floor(6 * self::randomFloat(0, 1) * self::randomFloat(0, 1));
		return $pressureTelemetryValue;
	}

	public function popNextPressurePsiValue() 
	{
		$pressureTelemetryValue = self::SamplePressure();
		return self::OFFSET + $pressureTelemetryValue;
	}
	
	private static function randomFloat($min = 0, $max = 1)
	{
		return $min + mt_rand() / mt_getrandmax() * ($max - $min);
	}
}

File: TDDMicroExercises/CSharp/TirePressureMonitoringSystem/Sensor.cs
using System;

namespace TDDMicroExercises.TirePressureMonitoringSystem
{
    public class Sensor
    {
        //
        // The reading of the pressure value from the sensor is simulated in this implementation.
        // Because the focus of the exercise is on the other class.
        //

        const double Offset = 16;
        readonly Random _randomPressureSampleSimulator = new Random();

        public double PopNextPressurePsiValue()
        {
            double pressureTelemetryValue = ReadPressureSample();

            return Offset + pressureTelemetryValue;
        }

        private double ReadPressureSample()
        {
            // Simulate info read from a real sensor in a real tire
            return 6 * _randomPressureSampleSimulator.NextDouble() * _randomPressureSampleSimulator.NextDouble();
        }
    }
}
